hr_course/models/hr_course.py

```diff
@@ -180,6 +183,48 @@ def cancel_course(self):
         for record in self:
             record.write(record._cancel_course_values())
 
+    @api.model
+    def send_course_notification_email(self):
+        company_id = self.env.context.get("company_id") or self.env.company.id
+        channel = (
+            self.env["res.company"].browse(company_id).course_expiration_channel_id
+        )
+        email_template = self.env.ref("hr_course.mail_template_validity_reminder")
+
+        if channel and self._name == "hr.course":
+            channel.with_context(
+                course_attendee_ids=self.course_attendee_ids,
+                recipient_ids=channel.channel_last_seen_partner_ids.mapped(
+                    "partner_email"
+                ),
+                email_from=channel.alias_id.display_name,
+            ).message_post_with_template(
+                template_id=email_template.id, res_id=self.id, model="hr.course"
+            )
```

Comment:
if you choose send params through context then why is there is still ${object.name} in the template
if there is res_id I expect that object is set respectively and no need to change mail.temlate 
check if res_id set correctly in the template through the method

Reply:
object = hr.course
agree that course_attendee_ids isn't needed in context but we still need reciptients and email_from - both coming from channel.